### ISSUE TITLE
Add examples covering all major aspects of the spec

### ISSUE THREAD

**Issue by chrisfilo** [bids-standard/bids-examples#128]

**Comment by Remi-Gau** [comment]
I suspect this fits in this issue.

Would there be some interest in "augmenting" the synthetic dataset to make it cover more aspects of the specs:
- json files covering most or all the RECOMMENDED fields
- have example data of all modalities
- ensure that this dataset "tracks" the development of the specs and is updated accordingly.

I have found myself writing [scripts](https://gist.github.com/Remi-Gau/6d179e8b59ab9c985369cba2f7282e94) to create fake dummy BIDS data sets for testing and I figure that maybe I could tweak them to update this "synthetic" dataset "easily" (#FamousLastWords).

**Comment by sappelhoff** [comment]
as far as I know, @effigies prepared the synthetic dataset --> I think your request is reasonable, but perhaps the scope ("tracking the full development of the spec") is a little too large, and we could have a hand full of synthetic datasets that achieve this job together, supplementing each other.

**Comment by Remi-Gau** [comment]
I agree that I might tend to get carried away with that.

I do agree with the idea of smaller modular synthetic datasets supplementing each other.

I will evnetually get to this.

I might open a draft PR to get the general structure but maybe a smart way would be to rely on some of the "writing" capabilites of pybids (https://github.com/bids-standard/pybids/issues/646 that I need to start enhancing to reuse in another project).

Would that seem like an acceptable way to proceed?

**Comment by sappelhoff** [comment]
> Would that seem like an acceptable way to proceed?

for me yes, and thanks a lot for all your activity in the last weeks to steadily improve BIDS!

I'd like to hear Chris' view on this :)

**Comment by effigies** [comment]
Yes, I like the idea of a handful of synthetic datasets, rather than building one out larger.

That dataset specifically is designed to have variables that can be loaded to test construction of design matrices, by the way. I would try to be very clear on exactly what you're trying to build with each one.

**Comment by mkoculak** [comment]
Hi, I am writing code to interact with BIDS data and was looking for some "mostly complete" datasets to test my implementations. I am using this repo now, but most of the datasets are closer to minimal implementation.
Are there any such datasets you can point me to?
If not, this is is an issue bump, as it would be great to have something like this. :)

**Comment by sappelhoff** [comment]
I think the idea of bids-examples so far was to provide self-contained, small examples that showcase specific ideas, rather than having huge monolithic examples.

If you want a BIDS dataset that covers as much as possible (several modalities, derivatives, lots of optional files, bids-ignored files, ...) then perhaps you could checkout some real-life datasets on openneuro?

**Comment by mkoculak** [comment]
Yes, I'll probably go for that in some time.
And I did notice bids-examples are more of a showcase, but since this issue was started 5 years ago, I hoped maybe a more broad example appeared somewhere.
This also means that all bids software is tested only on these small examples? So is it safe to assume that every major aspect of the spec is present at least in one of the datasets?

**Comment by sappelhoff** [comment]
> This also means that all bids software is tested only on these small examples?

No, "BIDS software" is a broad term, and all software devs decide about their own testing procedures. This may or may not include the examples from this "bids-examples" repository, but in most cases softwares will include additional tests.

>  So is it safe to assume that every major aspect of the spec is present at least in one of the datasets?

That's the goal, but it's not safe to assume that. Perhaps we should do a survey and try to guarantee that for users.

**Comment by robertoostenveld** [comment]
> Perhaps we should do a survey and try to guarantee that for users.

I don't think we are be able to provide this or even get close. There is an exponentially exploding number of possibilities in BIDS datasets and it is not possible to make examples all of them. For example considering the non-anticipated use-case of iEEG/LFP with MEG as discussed in #1550: there could be MEG with iEEG in the same file, MEG with EEG in the same file, EEG with both iEEG and EEG in the same file, MEG with iEEG in a different file, MEG with EEG in a different file, etc..., and each of them could or could not have electrode positions specified.
